crypto/ui/ui_lib.c

```diff
@@ -43,7 +43,7 @@ UI *UI_new_method(const UI_METHOD *method)
     ret->meth = method;
 
     if (!CRYPTO_new_ex_data(CRYPTO_EX_INDEX_UI, ret, &ret->ex_data)) {
-        OPENSSL_free(ret);
+        UI_free(ret);
```

Comment:
Is this safe to call CRYPTO_free_ex_data here?

Reply:
@levitte [thought it probably was](https://github.com/openssl/openssl/pull/15834#discussion_r657671665). 

I can't see anything wrong either -- it's NULL checking properly.